assets/src/editor/events/tickets/editor-ticket/filter-bar/with-ticket-entities-list-filter-bar.js

```diff
@@ -42,80 +47,98 @@ export const getFilteredTicketEntitiesList = ( ticketEntities, showTickets, tick
-		return class extends Component {
-			render() {
-				const {
-					showTickets,
-					setShowTickets,
-					ticketsSortedBy,
-					setTicketsSortedBy,
-					displayTicketDate,
-					setDisplayTicketDate,
-					isChained,
-					setIsChained,
-					searchTicketName,
-					setSearchTicketName,
-					ticketsPerPage,
-					setTicketsPerPage,
-					ticketsView,
-					setTicketsListView,
-					setTicketsGridView,
-					prefiltered = false,
+const withTicketEntitiesListFilterBar = createHigherOrderComponent(
+	( EntityList ) => ( {
+		showTickets,
+		setShowTickets,
+		ticketsSortedBy,
+		setTicketsSortedBy,
+		displayTicketDate,
+		setDisplayTicketDate,
+		isChained,
+		setIsChained,
+		searchTicketName,
+		setSearchTicketName,
+		ticketsPerPage,
+		setTicketsPerPage,
+		ticketsView,
+		setTicketsListView,
+		setTicketsGridView,
+		setFilteredTicketEntities,
+		ticketEntities = DEFAULT_EMPTY_ARRAY,
+		filteredDateEntities = DEFAULT_EMPTY_ARRAY,
+		...otherProps
+	} ) => {
+		const filteredEntities = useMemo(
+			() => {
+				const entities = searchTicketEntities( ticketEntities, searchTicketName );
+				return getFilteredTicketEntitiesList(
 					entities,
-					...otherProps
-				} = this.props;
-				let filteredEntities = searchTicketEntities( entities, searchTicketName );
-				filteredEntities = prefiltered ?
-					filteredEntities :
-					getFilteredTicketEntitiesList(
-						filteredEntities,
-						showTickets,
-						ticketsSortedBy
-					);
-				return (
-					<Fragment>
-						<EntityListFilterBar
-							name="TicketEntityListFilterBar"
-							searchText={ searchTicketName }
-							setSearchText={ setSearchTicketName }
-							perPage={ ticketsPerPage }
-							view={ ticketsView }
-							setPerPage={ setTicketsPerPage }
-							setListView={ setTicketsListView }
-							setGridView={ setTicketsGridView }
-							entityFilters={
-								<TicketEntityListFilterBar
-									showTickets={ showTickets }
-									setShowTickets={ setShowTickets }
-									ticketsSortedBy={ ticketsSortedBy }
-									setTicketsSortedBy={ setTicketsSortedBy }
-									displayTicketDate={ displayTicketDate }
-									setDisplayTicketDate={ setDisplayTicketDate }
-									isChained={ isChained }
-									setIsChained={ setIsChained }
-								/>
-							}
-						/>
-						<EntityList
-							entities={ filteredEntities }
-							entitiesPerPage={ ticketsPerPage }
-							view={ ticketsView }
-							noResultsText={
-								__(
-									'no results found (try changing filters)',
-									'event_espresso'
-								)
-							}
+					showTickets,
+					ticketsSortedBy
+				);
+			},
+			[
+				ticketEntities,
+				searchTicketName,
+				showTickets,
+				ticketsSortedBy,
+			]
+		);
+		// whenever filtered entities changes let's update the ticket ids
+		// in the state.
+		useEffect( () => {
+			setFilteredTicketEntities(
+				filteredEntities.map( ( ticketEntity ) => ticketEntity.id )
+			);
+		}, [ isChained, filteredDateEntities ] );
```

Comment:
What this logic indicates is that there is only need to update the filtered ticket entities (ids)  in the state if either `isChained` value changes or if `filteredDateEntities` changes.  However, this may need revisited because this means the state will not get updated if the ticket entity is not chained and filter selections are updated for tickets.

I have this the way it is currently because `ticketEntities` without this, the effect fires really frequently (add a console.log inside the effect).  That may have been just something in an earlier iteration of this pull and won't happen now but we probably should have this also include any of the filter state toggles as dependencies too.

Reply:
i checked out the branch and played around with the filters and everything seems to work ok (except for a bug in the sort chronologically filter but that has nothing to do with this branch)

Reply:
> except for a bug in the sort chronologically filter but that has nothing to do with this branch

Is that bug documented anywhere?

Reply:
no but I will be making a pull after this gets merged